lib/c/libintvector.h

```diff
@@ -643,30 +656,34 @@ static inline Lib_IntVector_Intrinsics_vec128 Lib_IntVector_Intrinsics_vec128_lo
 #define Lib_IntVector_Intrinsics_vec128_interleave_high64(x1,x2) \
   (vreinterpretq_u32_u64(vzip2q_u64(vreinterpretq_u64_u32(x1),vreinterpretq_u64_u32(x2))))
 
-// IBM z architecture
-#elif defined(__s390x__) // this flag is for GCC only
+#endif /* EVERCRYPT_CAN_COMPILE_VEC128 */
+
+/* IBM z architecture */
+#elif defined(__s390x__) /* this flag is for GCC only */
+
+#if defined(EVERCRYPT_CAN_COMPILE_VEC128)
 
 #include <vecintrin.h>
 
-// The main vector 128 type
-// We can't use uint8_t, uint32_t, uint64_t... instead of unsigned char,
-// unsigned int, unsigned long long: the compiler complains that the parameter
-// combination is invalid.
+/* The main vector 128 type
```

Comment:
Unrelated but because you changed the comment style here. IIRC `//` comments are only allowed starting C99, i.e. the C89 version shouldn't compile 🤔

Reply:
I was hoping to make this file compile in C89 mode since that's a property we're trying to preserve?

Reply:
Yes, it's a good chance. I was only wondering why no one complained about the c89 dist not actually being c89 compatible 🙂

Reply:
apparently you need -ansi to trigger the rejection of C++-style comments -- we don't use this flag. I was able to confirm that the command-line features -std=c89 -- I seem to recall some other clients of HACL downstream *do* use more pendantic modes, so perhaps it's a good thing to do to try to maintain C89-style comments in any case.